source/applications/advanced/gamma_correction.py

```diff
@@ -0,0 +1,89 @@
+"""
+Capture a 2D image and apply a configurable gamma correction.
+"""
+
+import argparse
+import numpy as np
+import cv2
+import zivid
+
+
+def _options():
+    """Configure and take command line arguments from user
+
+    Returns:
+        Arguments from user
+
+    """
+    parser = argparse.ArgumentParser(
+        description=(
+            "Capture 2D image and apply gamma correction\n"
+            "Example:\n\t $ python gamma_correction.py 2"
+        ),
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    parser.add_argument(
+        "gamma", nargs="?", type=float, help="Gamma correction value", default="2"
+    )
+
+    return parser.parse_args()
+
+
+def adjust_gamma(image, gamma: float):
+    """Adjust gamma on image.
+
+    Args:
+        image: OpenCV image to be corrected
+        gamma: Gamma setting to apply
+
+    Returns:
+        Gamma adjusted (OpenCV) image
+
+    """
+    # build a lookup table mapping the pixel values [0, 255] to
+    # their adjusted gamma values
+    inv_gamma = 1.0 / gamma
+    table = np.array(
+        [((i / 255.0) ** inv_gamma) * 255 for i in np.arange(0, 256)]
+    ).astype("uint8")
+    # apply gamma correction using the lookup table
+    return cv2.LUT(image, table)
+
+
+def _capture_2d_image(camera):
+    """Capture and extract 2D image, then convert from RGBA and return BGR.
+
+    Args:
+        camera: Zivid Camera handle
+
+    Returns:
+        OpenCV BGR image
+
+    """
+    settings_2d = zivid.Settings2D()
+    with camera.capture_2d(settings_2d) as frame_2d:
+        image = frame_2d.image()
+        image_array = image.to_array()
+        return np.dstack([image_array["b"], image_array["g"], image_array["r"]])
+
+
+def _main():
+    app = zivid.Application()
+
+    user_options = _options()
+
+    camera = app.connect_camera()
+
+    bgr_original = _capture_2d_image(camera)
+    cv2.imwrite("original.jpg", bgr_original)
```

Comment:
I would display both images, before and after:
`    cv2.imshow("Images", np.hstack([bgr_original]))
    cv2.waitKey(0)`

Reply:
And add a title to it, or print out displaying this image.

Reply:
Same for the one afte gamma.

Reply:
yeah, I agree. I even thought about splitting the image half way, and show half the image original and half corrected. Won't do it if the code is too complicated